### DISCUSSION TITLE
Access the correlation functions during training

### DISCUSSION THREAD

**Discussion by S3anaaaaz** [cornellius-gp/gpytorch#1988] [Q&A]
Hi, how can I access to the different parts of a GP model in exactgp like covariance matrix and mean for calculating the LOOCV?

Thanks,

Sanaz

**Comment by wjmaddox** [comment]
If you have a standard GP model (e.g. like in https://docs.gpytorch.ai/en/stable/examples/01_Exact_GPs/Simple_GP_Regression.html), then a forwards pass from the model will give you the mean and covariance:

```python
model = ExactGP(train_x, train_y, likelihood)

output = model(train_x)
output.mean # prior mean
output.covariance_matrix # prior covariance matrix
```

But more generally, the LOOCV is already implemented in gpytorch, see https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/mlls/leave_one_out_pseudo_likelihood.py (not sure why it's not coming up in the docs).

```python
from gpytorch.mlls import LeaveOneOutPseudoLikelihood

loocv = gpytorch.mlls.LeaveOneOutPseudoLikelihood(likelihood, model)
output = model(train_x)
loss = -loocv(output, train_y)
```
